internal/twdesk/priorities.go

```diff
@@ -101,82 +144,112 @@ func PriorityList(client *deskclient.Client) server.ServerTool {
+					},
+				},
+				Required: []string{"name"},
+			},
+		},
+		Handler: func(ctx context.Context, request *mcp.CallToolRequest) (*mcp.CallToolResult, error) {
+			arguments, err := helpers.NewToolArguments(request)
+			if err != nil {
+				return helpers.NewToolResultTextError(err.Error()), nil
+			}
+
 			priority, err := client.TicketPriorities.Create(ctx, &deskmodels.TicketPriorityResponse{
 				TicketPriority: deskmodels.TicketPriority{
-					Name:  request.GetString("name", ""),
-					Color: request.GetString("color", ""),
+					Name:  arguments.GetString("name", ""),
+					Color: arguments.GetString("color", ""),
 				},
 			})
 			if err != nil {
 				return nil, fmt.Errorf("failed to create priority: %w", err)
 			}
 
-			return mcp.NewToolResultText(fmt.Sprintf("Priority created successfully with ID %d", priority.TicketPriority.ID)), nil
+			id := priority.TicketPriority.ID
+			return helpers.NewToolResultText(fmt.Sprintf("Priority created successfully with ID %d", id)), nil
 		},
 	}
 }
 
 // PriorityUpdate updates a priority in Teamwork Desk
-func PriorityUpdate(client *deskclient.Client) server.ServerTool {
-	return server.ServerTool{
-		Tool: mcp.NewTool(string(MethodPriorityUpdate),
-			mcp.WithTitleAnnotation("Update Priority"),
-			mcp.WithDescription(
-				"Update an existing priority in Teamwork Desk by ID, allowing changes to its name and color. "+
-					"Supports evolving support policies, rebranding, or correcting priority attributes for improved "+
-					"ticket handling."),
-			mcp.WithNumber("id",
-				mcp.Required(),
-				mcp.Description("The ID of the priority to update."),
-			),
-			mcp.WithString("name",
-				mcp.Description("The new name of the priority."),
-			),
-			mcp.WithString("color",
-				mcp.Description("The color of the priority."),
-			),
-		),
-		Handler: func(ctx context.Context, request mcp.CallToolRequest) (*mcp.CallToolResult, error) {
-			_, err := client.TicketPriorities.Update(ctx, request.GetInt("id", 0), &deskmodels.TicketPriorityResponse{
+func PriorityUpdate(client *deskclient.Client) toolsets.ToolWrapper {
+	return toolsets.ToolWrapper{
+		Tool: &mcp.Tool{
+			Name: string(MethodPriorityUpdate),
+			Annotations: &mcp.ToolAnnotations{
+				Title: "Update Priority",
+			},
+			Description: "Update an existing priority in Teamwork Desk by ID, allowing changes to its name and color. " +
+				"Supports evolving support policies, rebranding, or correcting priority attributes for improved " +
+				"ticket handling.",
+			InputSchema: &jsonschema.Schema{
+				Type: "object",
+				Properties: map[string]*jsonschema.Schema{
+					"id": {
+						Type:        "integer",
+						Description: "The ID of the priority to update.",
+					},
+					"name": {
+						Type:        "string",
+						Description: "The new name of the priority.",
+					},
+					"color": {
+						Type:        "string",
+						Description: "The color of the priority.",
+					},
+				},
+				Required: []string{"id"},
```

Comment:
That's kind of annoying now.  Almost have to use constants to ensure field names are the same.

Reply:
I will hold merging this PR until we resolve this discussion, in case you have some good workaround ideas.

Reply:
The only alternative was to create constants.  It's probably not a big deal though.  Just keep it this way for now and once they resolve we can fix it.